src/presentation/web/src/components/registerForm/RegisterForm.tsx

```diff
@@ -0,0 +1,210 @@
+import { useMutate } from 'restful-react';
+import { KeyboardDatePicker } from '@material-ui/pickers';
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import { Link as RouterLink, Redirect } from 'react-router-dom';
+
+interface Inputs {
+    name: string;
+    surname: string;
+    mail: string;
+    password: string;
+    repPassword: string;
+    birthDate: Date;
+    phone: number;
+}
+const RegisterForm: React.FC = () => {
+    const theme = useTheme<Theme>();
+    const useStyle = makeStyles({
+        lockBackground: {
+            backgroundColor: theme.palette.secondary.dark,
+            borderRadius: 90,
+            padding: 8,
+            marginBottom: 10
+        },
+        lockIcon: {
+            color: '#FFF',
+            opacity: .87,
+        },
+        submit: {
+            filter: 'drop-shadow(0px 3px 1px rgba(0, 0, 0, 0.2)), drop-shadow(0px 2px 2px rgba(0, 0, 0, 0.14)), drop-shadow(0px 1px 5px rgba(0, 0, 0, 0.12))',
+            marginBottom: 10
+        },
+        paper: {
+            color: theme.palette.background.paper,
+            margin: '3rem 0',
+            padding: '20px 50px',
+            display: 'flex',
+            flexDirection: 'column',
+            justifyContent: 'space-between',
+            alignItems: 'center'
+        },
+        text: {
+            color: theme.palette.text.primary,
+            marginBottom: 35
+        },
+        link: {
+            color: theme.palette.info.dark,
+            alignSelf: 'flex-end',
+        },
+        textField: {
+            marginBottom: 35,
+        }
+    })
+    const classes = useStyle();
+    const emailPattern = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    const { register, handleSubmit, setError, errors, getValues, setValue, formState, trigger } = useForm<Inputs>({})
+
+    const [fireRedirect, setFireRedirect] = useState<boolean>(false);
+    const [date, setDate] = useState(new Date());
+    const handleDateChange = (date: Date | null): void => {
+        date && setDate(date);
+        setValue("birthDate", date);
+      };
+    const { mutate } = useMutate({
+        verb: 'POST',
+        path: '/users'
+    })
+
+    const onSubmit = async ({name, surname, mail, password, repPassword, birthDate, phone}: Inputs) => {
+        try {
+            await mutate({
+                name,
+                surname,
+                mail,
+                password,
+                repPassword,
+                birthDate: new Date(birthDate).toISOString(),
+                phone
+            })
+            setFireRedirect(true);
+        } catch (error) {
+            if (error.data.status === 400) setError('mail', { message: error.data.message })
+        }
+    }
+
+    const validateRepeatPassword = () => {
+        if (formState.isSubmitted) trigger('repPassword')
```

Comment:
Skomplikowana logika tego komponentu. Poświęciłem kilkanaście minut na ogarnięcie i parę razy przejrzałem kod i z tego co rozumiem to register() można także wywołać poza propsem komponentu w useEffect. A trigger() wywołuje ten register z useEffect. Jeśli to tak działa to cała logika się wydaje w porządku więc daje approve :)